src/utils/decoder.ts

```diff
@@ -0,0 +1,44 @@
+import { decode as decodeJwt } from "@pagopa/io-react-native-jwt";
+import type { JWTDecodeResult } from "@pagopa/io-react-native-jwt/lib/typescript/types";
+import { ValidationFailed } from "./errors";
+
+/*
+ * Decode a form_post.jwt and return the final JWT.
+ * The formData here is in form_post.jwt format as defined in
+ * JWT Secured Authorization Response Mode for OAuth 2.0 (JARM)
+ * HTTP/1.1 200 OK
+ *   Content-Type: text/html;charset=UTF-8
+ *   Cache-Control: no-cache, no-store
+ *   Pragma: no-cache
+ *
+ *   <html>
+ *   <head><title>Submit This Form</title></head>
+ *   <body onload="javascript:document.forms[0].submit()">
+ *     <form method="post" action="https://client.example.com/cb">
+ *       <input type="hidden" name="response"
+ *       value="eyJhbGciOiJSUz....."/>
+ *       </form>
+ *    </body>
+ *   </html>
+ */
+export const getJwtFromFormPost = async (
+  formData: string
+): Promise<{ jwt: string; decodedJwt: JWTDecodeResult }> => {
+  const formPostRegex = /<input(.|\n)*value\s*=\s*"((.|\n)*)"(.|\n)*>/gm;
+  const lineExpressionRegex = /\r\n|\n\r|\n|\r|\s+/g;
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** It would be great to have a test covering the scenario of those characters being trimmed. ‏

Reply:
Unit test is [here](https://github.com/pagopa/io-react-native-wallet/pull/47/files#diff-43df75678a472f5960db435df13e70167d096dca18db5c9fe53f69a2d6e1033d). Do you think it needs to be grittier?

Reply:
I don't know, maybe one more example showing the need for the replace? Just to stress the utility a bit on edge cases.

Reply:
the replace of new lines is also tested in the unit test